In [321]:
import pandas as pd

## Questions 1 - 3

which belt trains the most minutes per session?

In [322]:
bjj = pd.read_csv('bjj.csv')

In [323]:
bjj.shape

(9, 4)

In [324]:
bjj.info()

<class 'pandas.DataFrame'>
RangeIndex: 9 entries, 0 to 8
Data columns (total 4 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   member_id  9 non-null      int64  
 1   belt       9 non-null      str    
 2   minutes    8 non-null      float64
 3   rolls      8 non-null      float64
dtypes: float64(2), int64(1), str(1)
memory usage: 420.0 bytes


In [325]:
bjj.head()

,member_id,belt,minutes,rolls
0,1,blue,90.0,5.0
1,2,white,60.0,3.0
2,3,blue,90.0,6.0
3,4,purple,120.0,8.0
4,5,red,180.0,12.0


Key column would be BELT

## Questions 4 - 6

In [326]:
print('before:', bjj.shape[0], 'rows')

before: 9 rows


In [327]:
bjj_clean = bjj.dropna()

In [328]:
print('after: ', bjj_clean.shape[0], 'rows')

after:  7 rows


In [329]:
print('lost: ', bjj.shape[0] - bjj_clean.shape[0])

lost:  2


In [330]:
bjj_clean

,member_id,belt,minutes,rolls
0,1,blue,90.0,5.0
1,2,white,60.0,3.0
2,3,blue,90.0,6.0
3,4,purple,120.0,8.0
4,5,red,180.0,12.0
5,6,white,60.0,3.0
7,8,black,200.0,4.0


I decided to use dropna, because with how few values there are available, filling with average would affect how accurate results are. E.g. minute average per belt

## Questions 7 - 9

In [331]:
# Goal: answer my question — average minutes per belt, most first
# I expect 5 rows back, because there are 5 different belts in my data

answer = (bjj_clean
.groupby('belt')['minutes']

.agg(['mean', 'count'])

.sort_values('mean', ascending=False))

In [332]:
print(answer.shape)

(5, 2)


In [333]:
print(answer)

         mean  count
belt                
black   200.0      1
red     180.0      1
purple  120.0      1
blue     90.0      2
white    60.0      2


All belts results are based upon very little data. Therefore potentially being inacurate with more students' training records being written down

## Questions 10 - 12

In [338]:
# Goal: attach each member's belt to their sessions, and check nothing vanished

merged = pd.merge(bjj_clean, bjj, on='member_id', how='left')

print(bjj_clean.shape[0], '->', merged.shape[0]) # these should match

7 -> 7


In [344]:
merged['belt_x'].isna().sum()

np.int64(0)

## Questions 13 - 15

According to results based upon data provided, given that these results came from single record. Black belt trained the most (200 min), then red belt with 100min, purple with 120. Blue and white belts are consistent of two records compared to the one previously mentioned. Blue belts trained 90 min per session and white trained 60 min